File: allama/python-client/allama_client/exceptions.py
"""
Allama client exceptions
"""


class AllamaError(Exception):
    """Base exception for Allama client errors"""
    pass


class AllamaConnectionError(AllamaError):
    """Connection error when communicating with Allama server"""
    pass


class AllamaTimeoutError(AllamaError):
    """Timeout error when request takes too long"""
    pass


class AllamaAPIError(AllamaError):
    """API error returned by Allama server"""
    
    def __init__(self, message: str, status_code: int = None, response: dict = None):
        self.message = message
        self.status_code = status_code
        self.response = response
        super().__init__(message)

File: allama/js-client/src/exceptions.ts
/**
 * Allama client exceptions
 */

export class AllamaError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'AllamaError';
  }
}

export class AllamaConnectionError extends AllamaError {
  constructor(message: string) {
    super(message);
    this.name = 'AllamaConnectionError';
  }
}

export class AllamaTimeoutError extends AllamaError {
  constructor(message: string) {
    super(message);
    this.name = 'AllamaTimeoutError';
  }
}

export class AllamaAPIError extends AllamaError {
  public statusCode?: number;
  public response?: any;

  constructor(message: string, statusCode?: number, response?: any) {
    super(message);
    this.name = 'AllamaAPIError';
    this.statusCode = statusCode;
    this.response = response;
  }
}
